src/Components/Blueprints/BuildImagesButton.tsx

```diff
@@ -1,36 +1,138 @@
-import React from 'react';
+import React, {useState} from 'react';
 
-import { Button, ButtonProps } from '@patternfly/react-core';
+import {
+  ButtonProps,
+  Dropdown,
+  MenuToggle,
+  MenuToggleElement,
+  Menu,
+  MenuContent,
+  MenuList,
+  MenuItem, Flex, FlexItem, Spinner, MenuToggleAction,
+} from '@patternfly/react-core';
 
 import { selectSelectedBlueprintId } from '../../store/BlueprintSlice';
 import { useAppSelector } from '../../store/hooks';
-import { useComposeBlueprintMutation } from '../../store/imageBuilderApi';
+import {
+  ImageTypes,
+  useComposeBlueprintMutation,
+  useGetBlueprintQuery,
+} from '../../store/imageBuilderApi';
+import {EditSaveButton} from "../CreateImageWizardV2/steps/Review/Footer/EditDropdown";
+import {skipToken} from "@reduxjs/toolkit/query";
 
-type BuildImagesButtonPropTypes = {
-  variant?: ButtonProps['variant'];
-  children?: React.ReactNode;
-};
-
-export const BuildImagesButton = ({
-  variant,
-  children,
-}: BuildImagesButtonPropTypes) => {
+export const BuildImagesButton = () => {
   const selectedBlueprintId = useAppSelector(selectSelectedBlueprintId);
+  const [selectedTargets, setSelectedTargets] =useState<ImageTypes[]>([]);
   const [buildBlueprint, { isLoading: imageBuildLoading }] =
     useComposeBlueprintMutation();
   const onBuildHandler = async () => {
-    selectedBlueprintId &&
-      (await buildBlueprint({ id: selectedBlueprintId, body: {} }));
+    debugger;
+    if (selectedBlueprintId) {
+      await buildBlueprint({
+        id: selectedBlueprintId,
+        body: {
+          image_types: selectedTargets
+        },
+      })};
+  };
+  const [selectedItems, setSelectedItems] = React.useState<number[]>([]);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const onToggleClick = () => {
+    setIsOpen(!isOpen);
+  };
+  const onSelect = (
+    _event: React.MouseEvent<Element, MouseEvent> | undefined,
+    value: string | number | undefined
+  ) => {
+    setIsOpen(false);
+  };
+    const {
+      data: blueprintDetails,
+    } =  useGetBlueprintQuery(selectedBlueprintId ? { id: selectedBlueprintId } : skipToken)
+    const blueprintImageType = blueprintDetails?.image_requests.map(
+        (image) => image.image_type
+    );
+
+  const onSelect2 = (
+      _event: React.MouseEvent<Element, MouseEvent> | undefined,
+      itemId: number | string | undefined
+  ) => {
+    const item = itemId as number;
```

Comment:
We should stay away from type assertion whenever possible as those only allow space for mistakes and future bugs.

The prop types say that `itemId` can be number, string or undefined, but then the const says it's always number. There is no exception handling so if itemId returns as undefined, the service could crash as this line would try to index with undefined:
```
const imageType = blueprintImageType?.[item];
```

But that being said I agree with Ondřej that we could and probably should index with the imageType, making the type string and going from there :)